src/main/java/software/amazon/awssdk/crt/http/Http2ClientConnection.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.awssdk.crt.http;
+
+import software.amazon.awssdk.crt.CrtRuntimeException;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This class wraps aws-c-http to provide the basic HTTP/2 request/response
+ * functionality via the AWS Common Runtime.
+ *
+ * Http2ClientConnection represents a single connection to a HTTP/2 service
+ * endpoint.
+ *
+ * This class is not thread safe and should not be called from different
+ * threads.
+ */
```

Comment:
We should discuss this API in depth. It seems to me like this represents an H2 connection and not a stream.

From a usage perspective, this doesn't look like the right level of abstraction for Java or the other SDKs. I think the CRT should manage the stream aspect internally so that each SDK doesn't need to implement it in their own libraries.

i.e.  the return value of calling `connectionManager.acquireConnection()` should be an HTTP/2 stream under the hood and not a physical connection.

If a user wants to make 2 concurrent requests, they might do something like

```
HttpConnection conn1 = connectionManager.acquireConnection().join();
conn1.makeRequest(...).activate();

HttpConnection conn2 = connectionManager.acquireConnection().join();
conn1.makeRequest(...).activate();
```

It seems to me like in the current version of the code, this results in two HTTP/2 connections being used, rather than both requests being interleaved on on the same connection over two different streams.